src/saveload/misc_sl.cpp

```diff
@@ -39,7 +39,7 @@ void SaveViewportBeforeSaveGame()
 	if (w != nullptr) {
 		_saved_scrollpos_x = w->viewport->scrollpos_x;
 		_saved_scrollpos_y = w->viewport->scrollpos_y;
-		_saved_scrollpos_zoom = w->viewport->zoom;
+		_saved_scrollpos_zoom = UnScaleZoomGUI(w->viewport->zoom);
```

Comment:
I am not sure whether this scale/unscale in the saveload code makes sense.
If you are on ZOOM_LVL_OUT_32X before saving, this would try to save non-existing 64X.
If it is clamped to 32X in the savegame, then loading the game will scale it to 16X, so different that before.

I think this change of zoom-levels in save/load cycles is weird. Maybe it's better to just store/restore the zoom-level as-it, and not try to adjust the zoom level in case you switch the display between save/load?

Reply:
Saving non-existing 64X would be bad and the value would need to be clamped on load (if that's not done already). I think it would be good to store a GUI-zoom independent zoom level though, so save games load consistently. Indeed between different screens on the same machine, but also when save games are shared between players. I can split it off into a separate PR so it can be reviewed/discussed separately?

Reply:
I've split this off into #10007.